sdk/file_sdk/DefaultFormat/default_format.cpp

```diff
@@ -8,7 +8,15 @@ DefaultFormat::DefaultFormat(shared_ptr<IStream> file, const std::string& extens
 {
 }
 
-const vector<Tag>& DefaultFormat::ReadTags() {
+const vector<Tag> DefaultFormat::ReadTags() {
+  throw std::exception("not implemnted");
+}
+
+string DefaultFormat::GetOriginalExtension() {
+  throw std::exception("not implemnted");
```

Comment:
you can implement this by default to return the extension.
only in pfile it returns something else